### PR TITLE
GCS_MAVLink: ignore Clang out-of-range warning

### PR DESCRIPTION
**Pull request by OXINARF**
Clang gives the following warning:

```
warning: comparison of constant 4 with expression of type 'mavlink_channel_t' is always false [-Wtautological-constant-out-of-range-compare]
    if (chan >= MAVLINK_COMM_NUM_BUFFERS) {
        ~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~
```

We could remove the code because Clang is technically correct but _mavlink_channel_t_ is defined in the MAVLink project, which means it can be changed and this check starts to do some actual work. That is the reason I believe that ignoring the warning is the best option here.

### CODE REVIEW

File: libraries/GCS_MAVLink/GCS_MAVLink.h [hunk 1]
```diff
@@ -47,10 +47,14 @@ extern mavlink_system_t mavlink_system;
...
 static inline void comm_send_ch(mavlink_channel_t chan, uint8_t ch)
 {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wtautological-constant-out-of-range-compare"
     // sanity check chan
     if (chan >= MAVLINK_COMM_NUM_BUFFERS) {
         return;
     }
+#pragma clang diagnostic pop
+
```
Comment (guludo) [inline libraries/GCS_MAVLink/GCS_MAVLink.h]: I think it's a good idea to define an inline function for that.

**Comment by lucasdemarchi** [comment]
As @guludo noticed I think it would be better to factor out an inline function to check the number of channels. Then add the pragma in only one place. Also: https://semaphoreci.com/diydrones/ardupilot/branches/pull-request-3810/builds/1

You will need to update the build system so not to give errors on unknown pragmas

**Comment by OXINARF** [comment]
@guludo @lucasdemarchi I did the inline function and made other commits so that unknown pragmas don't get in the way.

**Comment by lucasdemarchi** [comment]
@OXINARF I did some changes on this pr and pushed for testing in the CI: https://github.com/lucasdemarchi/ardupilot/commits/review/pr-3810

  - coding style in control statements
  - We don't need to add a -Wno-error=<warn> if we are not emitting the warning (due to -Wno-<warn>)
  - squashed some commits changing the same lines
  - rename the function to be valid_channel - I don't think we gain much on abbreviating the function name
  - remove some comments that are now obvious due to the function name

Let me know if you agree with the changes and if CI is happy we can apply it.

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 2]
```diff
@@ -38,7 +38,7 @@ void
 GCS_MAVLINK::init(AP_HAL::UARTDriver *port, mavlink_channel_t mav_chan)
 {
     // sanity check chan
-    if (mav_chan >= MAVLINK_COMM_NUM_BUFFERS) {
+    if(!valid_chan(mav_chan)) {
```
Comment (lucasdemarchi) [inline libraries/GCS_MAVLink/GCS_Common.cpp]: according to the coding style, in control statements like "if", "for", etc there's a space before opening parenthesis, so I changed to `if (!valid_chann)`

Comment (lucasdemarchi) [inline libraries/GCS_MAVLink/GCS_Common.cpp]: and a link to the coding sytle document: http://ardupilot.org/dev/docs/style-guide.html

**Comment by OXINARF** [comment]
@lucasdemarchi:

- sorry about the coding style, forgot that rule
- I didn't know that just removing the warning would work when we had set Werror, always learning :+1: 
- you missed a comment in GCS_MAVLink.h :smile: 

Fine by me Lucas and thank you for your help!

**Comment by lucasdemarchi** [comment]
@OXINARF I fixed the issue mentioned and added a commit on top to simplify the function. Could you take a look again?

**Comment by OXINARF** [comment]
@lucasdemarchi It's fine Lucas, thanks!

**Comment by lucasdemarchi** [comment]
Applied, thanks

**Pull request closed without merging**